code/game/machinery/_machinery.dm

```diff
@@ -166,6 +166,14 @@ Class Procs:
 
 	return INITIALIZE_HINT_LATELOAD
 
+/obj/machinery/computer/can_interact(mob/user)
+	var/mob/living/living_user = user
+	if(ismonkey(living_user))
```

Comment:
So the thing is, there DOES exist a dexterity proc called IsAdvancedToolUser in addition to standard dexterity.

The thing is, its mostly for simplemobs.

For monkeys specifically, it doesn't really make sense for them not to do very basic tasks like press buttons, use a wrench, or be able to remove an embedded piece of glass from their foot.

There's really only 2 things I want to kneecap monkeys on, using computers and using certain weapons. Any check for what Im trying to do can't really be generic, since its really really specific.

If you think a trait will be advantageous, I will go ahead, but I don't really see the use.

Reply:
Generic code like traits is just always cleaner than having specific /typepath checks, I would always avoid checking specific paths in favour of using traits/components/overriding/anything generic where reasonable

Reply:
alright, Ill work something up for ashies & monkeys, and add code docs. Should be a decent genericization.

Reply:
Genericized in 0a30ff7